bundles/org.openhab.binding.ojelectronics/src/main/java/org/openhab/binding/ojelectronics/internal/services/RefreshService.java

```diff
@@ -0,0 +1,158 @@
+    }
+
+    /**
+     * Starts refreshing all thing values
+     *
+     * @param sessionId Session-Id
+     * @param refreshDone This method is called if refreshing is done.
+     * @param connectionLosed This method is called if no connection could established.
+     * @param unauthorized This method is called if the result is unauthorized.
+     */
+    public void start(String sessionId, Consumer<@Nullable GroupContentResponseModel> refreshDone,
+            Runnable connectionLost, Runnable unauthorized) {
+        logger.trace("RefreshService.startService({})", sessionId);
+        this.connectionLost = connectionLost;
+        this.refreshDone = refreshDone;
+        this.unauthorized = unauthorized;
+        this.sessionId = sessionId;
+        long refreshTime = config.refreshDelayInSeconds;
+        scheduler = schedulerService.scheduleWithFixedDelay(this::refresh, refreshTime, refreshTime, TimeUnit.SECONDS);
+        destroyed = false;
+    }
+
+    /**
+     * Stops refreshing.
+     */
+    public void stop() {
+        destroyed = true;
+        final ScheduledFuture<?> scheduler = this.scheduler;
+        if (scheduler != null) {
+            scheduler.cancel(false);
+        }
+        this.scheduler = null;
+    }
+
+    private void refresh() {
+        if (sessionId.equals("")) {
+            handleConnectionLost();
+            return;
+        }
+        final Runnable unauthorized = this.unauthorized;
+        createRequest().send(new BufferingResponseListener() {
+            @Override
+            public void onComplete(@Nullable Result result) {
+                if (!destroyed) {
+                    if (result == null || result.isFailed()) {
+                        handleConnectionLost();
+                    } else if (result.getResponse().getStatus() == HttpStatus.FORBIDDEN_403) {
+                        if (unauthorized != null) {
+                            unauthorized.run();
+                        }
+                    } else {
+                        handleRefreshDone(getContentAsString());
+                    }
+                }
+            }
+        });
+    }
+
+    private Request createRequest() {
+        Request request = httpClient.newRequest(config.apiUrl + "/Group/GroupContents").param("sessionid", sessionId)
+                .param("APIKEY", config.apiKey).method(HttpMethod.GET);
+        return request;
+    }
+
+    private void handleRefreshDone(String responseBody) {
+        final Consumer<@Nullable GroupContentResponseModel> refreshDone = this.refreshDone;
+        if (refreshDone != null) {
+            logger.trace("refresh {}", responseBody);
+            try {
+                GroupContentResponseModel content = gson.fromJson(responseBody, GroupContentResponseModel.class);
+                refreshDone.accept(content);
+            } catch (JsonSyntaxException exception) {
+                logger.error("Error mapping Result to model", exception);
```

Comment:
In [E.4 of the coding guidelines](http://docs.openhab.org/developers/development/guidelines.html#e-logging) we listed what kind of logging level should be used when. Error should only be used:

> `error` logging should only be used to inform the user that something is tremendously wrong in his setup, the system cannot function normally anymore, and there is a need for immediate action. 

Please change this to use the correct logging levels.

Reply:
I think the correct way is to remove the logging and disable the thing. Is this correct?

Reply:
Not really disabling, updating the thing status, calling `updateStatus` and I would see that this indicates a communication error, you could log the full stack trace to debug, but the most important is reporting to the user that something is wrong and passing the `exception.getMessage()` to the `updateStatus` could help for that.

Reply:
Done